bgpd/bgp_mplsvpn.c

```diff
@@ -1988,9 +1988,15 @@ void vpn_handle_router_id_update(struct bgp *bgp, bool withdraw,
 			bgp->vpn_policy[afi].tovpn_rd = bgp->vrf_prd_auto;
 			prefix_rd2str(&bgp->vpn_policy[afi].tovpn_rd, buf,
 				      sizeof(buf));
-			bgp->vpn_policy[afi].rtlist[edir] =
-				ecommunity_str2com(buf,
-						   ECOMMUNITY_ROUTE_TARGET, 0);
+
+			/* free up pre-existing memory if any and allocate
+			 *  the ecommunity attribute with new RD/RT
+			 */
+			if (bgp->vpn_policy[afi].rtlist[edir])
+				ecommunity_free(
+					&bgp->vpn_policy[afi].rtlist[edir]);
+			bgp->vpn_policy[afi].rtlist[edir] = ecommunity_str2com(
```

Comment:
Could you check if this is the same for:

```c
from_bgp->vpn_policy[afi].rtlist[edir] =
			ecommunity_str2com(buf, ECOMMUNITY_ROUTE_TARGET, 0);
```
as well?

Reply:
Sure. I had couple more leak fixes. I'll split it all in different commits.

Reply:
It might not be required here. 
In this case, we check if the source vrf is already configured to export to any other vrf. 
->  If not, only at the first export configuration for that vrf we are assigning the above. So, ecommunity will not be allocated before since it is the first instance.
-> If already configured to export, then above is not called.